boxsdk/auth/jwt_auth.py

```diff
@@ -360,3 +361,42 @@ def _normalize_rsa_private_key_passphrase(passphrase):
                 .format(passphrase.__class__.__name__)
             )
         return passphrase
+
+    @classmethod
+    def from_settings_dictionary(cls, config_dictionary, **kwargs):
+        """
+        Create an auth instance as defined by the given settings dictionary.
+
+        The dictionary should have the structure of the JSON file downloaded from the Box Developer Console.
+
+        :param config_dictionary:       Dictionary containing settings for configuring app auth.
+        :type config_dictionary:        `dict`
+        :return:                        Auth instance configured as specified by the config dictionary.
+        :rtype:                         :class:`JWTAuth`
+        """
+        if 'boxAppSettings' not in config_dictionary:
+            raise ValueError('boxAppSettings not present in configuration')
+        return cls(
+            client_id=config_dictionary['boxAppSettings']['clientID'],
+            client_secret=config_dictionary['boxAppSettings']['clientSecret'],
+            enterprise_id=config_dictionary.get('enterpriseID', None),
```

Comment:
This is different than `.get('enterpriseID') or None`. What you have here will return an empty string, if an empty string is given. What I had proposed will return `None`, if an empty string is given.

I know that our functions _sometimes_ treat an empty string the same as `None`, but I think it'd be safer and more correct to pass `None`.

Similarly below.

Reply:
The SDK right now does very little sanity checking. I don't think the empty string is a valid rsa passphrase (actually, I have no idea), and I also don't think it's a valid enterprise ID. But I'm not sure we need to special-case these by turning empty string into None.